packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx

```diff
@@ -44,52 +28,52 @@ export type QueryContextType<
   removeQueryFilter: ImmutableTableQueryResult['removeQueryFilter']
   /** Removes a value from a QueryFilter. If no more values remain in the filter, the filter is also removed */
   removeValueFromQueryFilter: ImmutableTableQueryResult['removeValueFromQueryFilter']
-  /** The error returned by the query request, if one is encountered */
-  error: SynapseClientError | null
-  /** The status of the asynchronous job. */
-  asyncJobStatus?: AsynchronousJobStatus<QueryBundleRequest, QueryResultBundle>
   /** Whether facets are available to be filtered upon based on the current data */
-  isFacetsAvailable: boolean
+  hasFacetedSelectColumn: boolean
   /** Returns true iff the current request has resettable filters applied via facet filters or additionalFilters. Excludes filters applied to a locked column */
   hasResettableFilters: boolean
-  getColumnModel: (columnName: string) => ColumnModel | null
   // Either open benefactor entity page in a new window or open the sharing settings dialog (in Synapse.org)
   onViewSharingSettingsClicked?: (benefactorId: string) => void
   /** Combines two faceted columns into a single inclusive range selector */
   combineRangeFacetConfig?: ReadonlyDeep<CombineRangeFacetConfig>
-}
 
-export type PaginatedQueryContextType<
-  TIncludedFields extends OptionalQueryBundleRequestFields = never,
-  ExcludeOtherFields extends true | false = false,
-> = QueryContextType<TIncludedFields, ExcludeOtherFields> & {
-  /** Navigates to a particular page, where the first page has value 1 */
+  /** react-query UseQueryOptions to fetch the current page of table data */
+  rowDataQueryOptions: TableQueryUseQueryOptions['rowDataQueryOptions']
+  /** react-query UseInfiniteQueryOptions to fetch infinite rows of table data */
+  rowDataInfiniteQueryOptions: TableQueryUseQueryOptions['rowDataInfiniteQueryOptions']
+  /** react-query UseQueryOptions to fetch the current query metadata (e.g. column models, facet statistics, total number of rows) */
+  queryMetadataQueryOptions: TableQueryUseQueryOptions['queryMetadataQueryOptions']
```

Comment:
For convenience, provide options which can be passed to `useQuery`, `useSuspenseQuery`, or `useInfiniteQuery` to get the data for the current query.